### PR TITLE
Dennis - Homework3

### PR DESCRIPTION
**Pull request by EvanSimpson**
Graded by @thomasnat1 @EvanSimpson

### CODE REVIEW

File: classes/class4/homework/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (EvanSimpson) [inline classes/class4/homework/package.json:1]: Did you mean to put your class 3 homework in your class 4 folder? Make sure you are careful of that in the future. Took us a minute to find the right application.

File: classes/class4/homework/routes/cats.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+module.exports = function(){
+    var express = require('express');
+    var router = express();
+    var Cat = require('../models/cat_model.js')
+
+    router.get('/', function(req,res,next) {
+      Cat.find({}).sort({age: -1}).exec(function(err,cats){
+        if (err) {
+            console.log('Error displaying all cats!');
+            res.status(404).send('Error displaying all cats!');
```
Comment (EvanSimpson) [inline classes/class4/homework/routes/cats.js:10]: Good error handling.

File: classes/class4/homework/routes/cats.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        function getRandomColors() {
+          var colors = ['blue','green','yellow','pink','purple','sunshine','cat-color'];
+          var random_colors = [];
+          for (var i=0; i < colors.length;i++){
+              if (Math.random() < 0.5){
+                random_colors.push(colors[i]);
+              }
+          }
+          if (random_colors.length === 0) {
+            random_colors = ['rainbowunicorn'];
```
Comment (EvanSimpson) [inline classes/class4/homework/routes/cats.js:38]: :+1: Nice easter egg.

File: classes/class4/homework/routes/cats.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    router.get('/', function(req,res,next) {
+      Cat.find({}).sort({age: -1}).exec(function(err,cats){
+        if (err) {
+            console.log('Error displaying all cats!');
+            res.status(404).send('Error displaying all cats!');
+        } else {
+            cats = cats.map(function (cat){
+                return 'name: '+cat['name']+' age: '+cat['age']+' colors: '+cat['colors'];
+            });
+            res.render('home',{'cats': cats});
```
Comment (ThomasTheDane) [inline classes/class4/homework/routes/cats.js:15]: You aren't actually using handlebars properly. You want to just send the cat array over and render the appropriate information in handlebars. Something like this: 

```
{{#each cats}}
  <li>{{name}}:
    <ul>
      <li>age: {{age}}</li>
      <li>color(s): {{#each colors}}{{this}}{{/each}}</li>
    </ul>
  </li>
  {{/each}}
```

File: classes/class4/homework/routes/cats.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }});
+    });
+            
+    router.get('/delete/old', function(req,res,next) {
+      Cat.find({}).sort({age: -1}).exec(function(err,cats){
+        if (err) {
+            console.log('Error deleting old cat!');
+            res.status(404).send('Error deleting old cat!');
+        } else {
+            to_remove = cats[0];
```
Comment (EvanSimpson) [inline classes/class4/homework/routes/cats.js:80]: Remember, always use `var` when creating a new variable.

File: classes/class4/homework/routes/cats.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+              if (Math.random() < 0.5){
+                random_colors.push(colors[i]);
+              }
+          }
+          if (random_colors.length === 0) {
+            random_colors = ['rainbowunicorn'];
+          }
+          return random_colors;
+        }
+        console.log(getRandomInt());
```
Comment (ThomasTheDane) [inline classes/class4/homework/routes/cats.js:42]: Having these logs isn't really too helpful without some other text (like console.log("made new cat of color " + color). I imagine they were for debugging but you should remove that when submitting.

File: classes/class4/homework/server.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+/*Routes with cats.js*/
+app.use('/cats', require('./routes/cats.js'));
```
Comment (EvanSimpson) [inline classes/class4/homework/server.js:25]: Avoid in-line `require`'s. 
Just put `var cat_routes = require('./routes/cats.js')` above, and use
`app.use('/cats', cat_routes)` down below.

File: classes/class4/homework/server.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+app.get('/test', function(req,res,next){
```
Comment (EvanSimpson) [inline classes/class4/homework/server.js:27]: Extra code. You don't really need this and can get rid of it before pushing to github.

File: classes/class4/homework/routes/cats.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
+module.exports = function(){
+    var express = require('express');
+    var router = express();
```
Comment (EvanSimpson) [inline classes/class4/homework/routes/cats.js:3]: :+1: Great use of the built-in router.

**Pull request closed without merging**